### PR TITLE
Update Storage SDK

### PR DESCRIPTION
**Pull request by esdrasjnr**
**Describe what this PR introduces:** This patch standardize Storage SDK with others SDK code

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce any breaking changes?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] Related issues are referenced in the PR (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included

**Testing environment:**

- Operating System/Platform: Ubuntu 18.04
- Node version: 12.15
- Yarn version: 1.22

If you have relevant logs, error output and etc, please attach them.

**Other information:**

### CODE REVIEW

File: .huskyrc.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 module.exports = {
   hooks: {
     'pre-commit': 'yarn lint',
-    'pre-push': 'yarn test',
```
Comment (netoax) [inline .huskyrc.js]: Were the tests working before the changes introduced by this PR?

File: package.json [hunk 2]
```diff
@@ -39,10 +39,10 @@
     "eslint-import-resolver-babel-module": "^5.0.1",
     "eslint-plugin-import": "^2.16.0",
     "eslint-plugin-prettier": "^3.1.4",
+    "husky": "^4.2.5",
```
Comment (netoax) [inline package.json:42]: What are the advantages of `husky`?

File: .huskyrc.js (hunk 1 above)
Comment (esdrasjnr) [reply .huskyrc.js]: No @netoax, I removed the old test code and this make this check fail. This is why I removed it! As you comment, maybe it's better to move this PR to draft until I add the appropriate test code. What do you think?

File: package.json (hunk 2 above)
Comment (esdrasjnr) [reply package.json:42]: In my researches, I found that it's more often used and have some additional hooks like `pre-push`. [Docs](https://github.com/typicode/husky)

Comment (netoax) [reply package.json:42]: Nice.

File: .huskyrc.js (hunk 1 above)
Comment (netoax) [reply .huskyrc.js]: I think if the tests weren't working before anyway, we can proceed with this PR.

File: src/network/HTTP.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import axios from 'axios';
+
+export default class HTTP {
+  constructor({ protocol = 'http', hostname = 'localhost', port = 8181, token }) {
```
Comment (netoax) [inline src/network/HTTP.js:4]:
```suggestion
  constructor({ protocol = 'http', hostname = 'localhost', port = 80, token }) {
```

File: src/network/HTTP.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    if (protocol !== 'http' && protocol !== 'https') {
+      throw new Error("Invalid protocol: must be either 'https' or 'http'");
+    }
+
+    this.baseUrl = `${protocol}://${hostname}:${port}`;
+    this.header = {
+      auth_token: token,
```
Comment (netoax) [inline src/network/HTTP.js:11]: Please, verify if this token is correctly provided.

File: src/network/HTTP.js (hunk 3 above)
Comment (netoax) [inline src/network/HTTP.js:4]: We should still support passing `pathname` parameter.

File: src/network/HTTP.js (hunk 4 above)
Comment (esdrasjnr) [reply src/network/HTTP.js:21]: Done!

File: src/network/HTTP.js (hunk 3 above)
Comment (esdrasjnr) [reply src/network/HTTP.js]: Done!

Comment (esdrasjnr) [reply src/network/HTTP.js]: Done! Also change the other params to the production cloud as default

File: .huskyrc.js (hunk 1 above)
Comment (esdrasjnr) [reply .huskyrc.js]: Yes, the old tests aren't passing :disappointed:

File: src/network/HTTP.js (hunk 3 above)
Comment (netoax) [reply src/network/HTTP.js]: If default is now https, the port must be changed to 443.

File: src/Storage.js [hunk 5]
```diff
@@ -1,105 +1,19 @@
...
-    );
-    const headers = {
-      auth_id: this.options.id,
-      auth_token: this.options.token,
-    };
-
-    const response = await axios.get(uri, { headers });
-    return response.data;
+  async listDataBySensor(thingId, sensorId, query) {
+    return this.http.get(`/data/$${thingId}/sensor/${sensorId}`, query);
```
Comment (netoax) [inline src/Storage.js:17]:
```suggestion
    return this.http.get(`/data/${thingId}/sensor/${sensorId}`, query);
```

Comment (esdrasjnr) [reply src/Storage.js:17]: Fix it!

File: src/network/HTTP.js (hunk 3 above)
Comment (esdrasjnr) [reply src/network/HTTP.js]: Done!

**Pull request merged**